### PR TITLE
fix(init): fix link and npm install

### PR DESCRIPTION
**Pull request by filipesilva**
Currently, `ng new proj --link-cli` breaks the linked `angular-cli` install due to https://github.com/npm/npm/issues/10343.

This PR changes our `init` process to link after install, which takes longer but doesn't break dependencies. It also does not use local `npm` and instead uses the global `npm` since it's what we already use for other commands.

/cc @hansl

### CODE REVIEW

File: packages/angular-cli/tasks/npm-install.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    let exitCode = 0;
+
+    return new Promise((resolve) => {
+      ui.writeLine(chalk.green('Installing packages for tooling via npm.'));
+      exec(`npm install`,
+        (err: NodeJS.ErrnoException, stdout: string, stderr: string) => {
+          if (err) {
+            ui.writeLine(stderr);
+            ui.writeLine(chalk.red('Package install failed, see above.'));
+            exitCode = 1;
```
Comment (hansl) [inline packages/angular-cli/tasks/npm-install.ts]: Any reason you're not rejecting the promise?

Comment (filipesilva) [inline packages/angular-cli/tasks/npm-install.ts]: Copy paste from e2e... which means that one is also broken.

Comment (filipesilva) [inline packages/angular-cli/tasks/npm-install.ts]: Fixed here but not the e2e. I assume it needs further changes and don't wanna mix the two things up.

**Comment by hansl** [comment]
Lgtm

**Comment by kylecordes** [comment]
This is already merged and needs no further look, but I just wanted to say, congratulations and thank you for moving away from the extra local install of in PM. That is a nice initial step toward paring back the dependency count and size.

**Pull request merged**